stl/inc/xtree

```diff
@@ -862,22 +863,18 @@ public:
     using reference       = value_type&;
     using const_reference = const value_type&;
 
-    using iterator =
-        conditional_t<is_same_v<key_type, value_type>, _Tree_const_iterator<_Scary_val>, _Tree_iterator<_Scary_val>>;
-    using const_iterator            = _Tree_const_iterator<_Scary_val>;
-    using _Unchecked_iterator       = conditional_t<is_same_v<key_type, value_type>,
-        _Tree_unchecked_const_iterator<_Scary_val>, _Tree_unchecked_iterator<_Scary_val>>;
+    using iterator       = conditional_t<_Is_set, _Tree_const_iterator<_Scary_val>, _Tree_iterator<_Scary_val>>;
+    using const_iterator = _Tree_const_iterator<_Scary_val>;
+    using _Unchecked_iterator =
+        conditional_t<_Is_set, _Tree_unchecked_const_iterator<_Scary_val>, _Tree_unchecked_iterator<_Scary_val>>;
     using _Unchecked_const_iterator = _Tree_unchecked_const_iterator<_Scary_val>;
 
     using reverse_iterator       = _STD reverse_iterator<iterator>;
     using const_reverse_iterator = _STD reverse_iterator<const_iterator>;
 
-    struct _Copy_tag {
-        explicit _Copy_tag() = default;
-    };
-
-    struct _Move_tag {
-        explicit _Move_tag() = default;
+    enum class _Copy_or_move_t {

```

Comment:
If there are only 2 alternatives we can explicitly use bool as base type. Also I would rather call this `_Strategy`
```suggestion
    enum class _Strategy : bool {
```

Reply:
What is the advantage?

Reply:
Its more explicit that it is a binary choice

Reply:
Interestingly, MSVC does not enforce it https://godbolt.org/z/EMY91hebv

Reply:
Never mind, it is still level 1 warning